src/core/utils/net.spec.ts

```diff
@@ -310,4 +310,13 @@ describe("net utilities", () => {
       expect(parseUrl("https://foo.com:9999")).toMatchObject({ port: 9999 });
     });
   });
+
+  describe("isHttpsUrl()", () => {
+    it("https url should be valid", () => {
+      expect(isHttpsUrl("https://foo.com")).toBeTrue();
+    });
+    it("http url should be invalid", () => {
+      expect(isHttpsUrl("http://foo.com")).toBeFalse();
+    });
```

Comment:
please also add invalid values to make sure the code won't throw unexpected errors.

Reply:
+1
Scenarios:
1. Null url
2. invalid url ( e..g random string  .. xysaa)
3. url with caps protocol - HTTPS4.